### ISSUE TITLE
Drop Down Field cmb2 call value in ajax

### ISSUE THREAD

**Issue by regidias2006** [WebDevStudios/CMB2#402]
Hello my name is Reginald, I have a doubt searched all over the internet and links imaginable excuse me my noob, but I'm in trouble I need an ajax call using the plugin cmb2, have values in a first drop and need to display the results in a second drop related to the first, I managed to make the ajax work but can not do so pull the values, it turns me returning the error message "NOT FOUND", can anyone help me?

my getvalue.php

function get_childlocations(){
   $parent_location_slug = $_POST['parent_location'];
   $parent_location_id = get_term_by('slug', $parent_location_slug, 'locations');
   $parent_location_id = $parent_location_id->term_id;
   
   $args = array(
            'taxonomy' => 'locations', 
            'type' => 'meeting',
            'parent' => $parent_location_id,
            'hide_empty' => 0
         );
   $child_locations = get_categories($args);
   $child_locations_array = array();

   foreach($child_locations as $child_location){
      $child_locations_array[] = array('child_slug' => $child_location->slug, 'child_name' => $child_location->name);
   }

   $child_locations_array = json_encode($child_locations_array);
   die($child_locations_array);
}

// create custom Ajax call for WordPress
add_action( 'wp_ajax_nopriv_get_childlocations', 'get_childlocations' );
add_action( 'wp_ajax_get_childlocations', 'get_childlocations' );

my callajax.js

jQuery("#property_location").change(function() {
	 var filter_location = jQuery(this).val();
		
    //clear options
		
		jQuery('#property_Cidades')
		.html('')
		jQuery('#property_Cidades').append('<option value="">carregando...</option>')
		.selectpicker('refresh');
		//reset the initial option
		jQuery('#property_Cidades')
		.html('')
		jQuery('#property_Cidades')
		.append('<option value="">Selecione uma cidade</option>');
		jQuery.ajax({
        type: 'POST',
        url: 'wp-admin/admin-ajax.php',
        data: {
            action: 'get_childlocations',
            parent_location: filter_location
        },
        dataType: 'json',
		cache:false,
        success: function(data, textStatus, XMLHttpRequest){
			
            //append results
            if(data){
					jQuery('#property_Cidades').selectpicker('refresh');
					jQuery.each(data, function(i,v) {   
                    jQuery('#property_Cidades')
						
                        .append(jQuery('<option>', { value : v.child_slug })
						.text(v.child_name));
					jQuery('#property_Cidades').selectpicker('refresh')	
						 
                });
            }
	      },
		
        error: function(MLHttpRequest, textStatus, errorThrown){
            alert(errorThrown);
        }
		
    });
	});

and my first select, this value is rendering in plugin cbm2 field type.php and cbm2.php

public function taxonomy_select() {

		$names      = $this->get_object_terms();
		$saved_term = is_wp_error( $names ) || empty( $names ) ? $this->field->args( 'default' ) : $names[key( $names )]->slug;
		$terms      = get_terms( $this->field->args( 'taxonomy' ), 'hide_empty=0' );
		$options    = '';

		$option_none  = $this->field->args( 'show_option_none' );
		if ( ! empty( $option_none ) ) {
			$option_none_value = apply_filters( 'cmb2_taxonomy_select_default_value', '' );
			$option_none_value = apply_filters( "cmb2_taxonomy_select_{$this->_id()}_default_value", $option_none_value );

			$options .= $this->select_option( array(
				'label'   => $option_none,
				'value'   => $option_none_value,
				'checked' => $saved_term == $option_none_value,
			) );
		}

		foreach ( $terms as $term ) {
			$options .= $this->select_option( array(
				'label'   => $term->name,
				'value'   => $term->slug,
				'checked' => $saved_term == $term->slug,
			) );
		}

		return $this->select( array( 'options' => $options ) );
	}

**Comment by JayWood** [comment]
@regidias2006 have a look at this simple script, should get you pointed in the right direction.
https://github.com/JayWood/cmb2-ajaxified-dropdown-snippet

Also, have a look at an article I wrote for working with AJAX requests: 
http://webdevstudios.com/2015/02/12/handling-ajax-in-wordpress/

Hope it helps!

**Comment by regidias2006** [comment]
@JayWood Thanks my friend i look your code, i go to study your code, when i finish my work i will show for all this code, i'm great so happy by you help me thanks.

**Comment by regidias2006** [comment]
hello, @JayWood thanks I studied your code and completely understand how it works, I could not call the metabox to my front end editor post because I got no way to call get_post_meta because it uses variables while I call the id and meta, you can help me?

**Comment by tnorthcutt** [comment]
Unfortunately, we can't support every person's specific or custom implementations for CMB2. Github issues are reserved for confirmed bug reports and occasional feature requests. Please review [CONTRIBUTING.md](https://github.com/WebDevStudios/CMB2/blob/master/CONTRIBUTING.md) for support suggestions. Please also review the additional resources listed at the bottom of that document, as the solution you're looking for may already be present.
